code/mob/living/critter/small_animal.dm

```diff
@@ -447,22 +456,114 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 		if (prob(10))
 			src.audible_message("[src] purrs!",\
 			"You purr!")
+		if (src.ai?.enabled && ispug(user) && prob(10))
+			ON_COOLDOWN(src, "recent_pug_pet", 15 SECONDS)
+			src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(src, src.ai.default_task))
+			src.ai.interrupt()
+			src.visible_message("<span class='notice'>[src] recoils and hisses at [user]'s attempt to pet them, then goes for the jugular!</span>")
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+
+	Crossed(atom/movable/M as mob)
+		..()
+		if (!isalive(src))
+			return
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(prob(10) && src.is_annoying)
+				src.visible_message("<span class='combat'>[src] weaves around [H]'s legs and trips [him_or_her(H)]!</span>")
+				H.setStatus("resting", duration = INFINITE_STATUS)
+				H.force_laydown_standup()
+			else if (prob(4) && ishuman(M))
+				boutput(src, "<span class='notice'>You weave around [H] to [pick("show your affection!", "get them to feed you.", "annoy them for no reason in particular.")]</span>")
+				boutput(H, "<span class='notice'>[src] weaves around you, waving their tail around. A bunch of hair clings to your clothes and some gets in your nose.</span>")
+				H.emote("sneeze")
+
+	seek_target(range)
+		. = list()
+		var/list/hearers_list = hearers(range, src)
+		for (var/mob/living/M in hearers_list)
+			if (istype(M, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = M
+				if (ispug(H) && isalive(H) && GET_COOLDOWN(src, "recent_pug_pet"))
+					. += H
+					return
+			if (istype(M, /mob/living/critter/small_animal/mouse))
+				var/mob/living/critter/small_animal/mouse/mouse = M
+				if (isdead(mouse)) continue
+				. += mouse
+		for (var/obj/critter/livingtail/tail in range(src, 3))
+			. += tail
+
+		if (length(.) && prob(20))
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+			src.visible_message("<span class='alert'>[src] hisses!</span>")
+
+	critter_attack(var/the_target)
+		if (istype(the_target, /obj/critter))
+			var/obj/critter/C = the_target
+			if (C.health <= 0 && C.alive)
+				playsound(src.loc, 'sound/impact_sounds/Generic_Hit_1.ogg', 50, 1, -1)
+				C.health -= 2
+		else if (istype(the_target, /mob))
+			var/mob/target = the_target
+			if(istype(target, /mob/living/critter/small_animal/mouse/weak/mentor) && prob(90))
+				src.visible_message("<span class='combat'><B>[src]</B> tries to bite [target] but \the [target] dodges [pick("nimbly", "effortlessly", "gracefully")]!</span>")
+				return
+			if ((src.catnip || prob(2) ) && (!GET_COOLDOWN(src, "claw_fury")))
+				ON_COOLDOWN(src, "claw_fury", 20 SECONDS)
```

Comment:
```suggestion
			if ((src.catnip || prob(2) ) && (!ON_COOLDOWN(src, "claw_fury", 20 SECONDS)))
```
`ON_COOLDOWN` already returns true/false based on the cooldown status and sets the cooldown if the thing wasn't on cooldown, so it does both these thing basically